das_api/src/api/mod.rs

```diff
@@ -204,6 +206,61 @@ pub struct GetTokenLargestAccounts(pub String, #[serde(default)] pub Option<Comm
 
 #[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, JsonSchema)]
 pub struct GetTokenSupply(pub String, #[serde(default)] pub Option<CommitmentConfig>);
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize, JsonSchema)]
+#[serde(deny_unknown_fields, rename_all = "camelCase")]
+pub struct GetTokenAccountOptionalParams {
+    #[serde(default)]
+    pub mint: Option<String>,
+    #[serde(default)]
+    pub program_id: Option<String>,
+}
+
+pub type GetTokenAccountOptionalParamsResult =
+    Result<(Option<Vec<u8>>, Option<Vec<u8>>), DasApiError>;
+
+impl GetTokenAccountOptionalParams {
+    pub fn validate(params: &Option<Self>) -> GetTokenAccountOptionalParamsResult {
+        if let Some(params) = params {
+            let mint = validate_opt_pubkey(&params.mint)?;
+            let program_id = validate_opt_token_program(&params.program_id)?;
+
+            return Ok((mint, program_id));
+        }
+
+        Ok((None, None))
+    }
+}
```

Comment:
If doing this then would expect validate to return bool. Then just use the instance of params you checked downstream.